### PR TITLE
Update TranscriptParser to use URLBuilder

### PR DESCRIPTION
**Pull request by n-wach**
Closes: #42

### CODE REVIEW

File: src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java [hunk 1]
```diff
@@ -78,16 +83,26 @@ public void parseAndStoreTranscript(String videoId, Key lectureKey) throws IOExc
...
   /**
    * Returns the transcript for a video as a document. Otherwise, returns Optional.empty()
-   * if there is a parsing error.
+   * if there is a parsing or URL building error.
    *
    * @param videoId Indicates the video to extract the transcript from.
    */
   private Optional<Document> getTranscriptXmlAsDocument(String videoId) throws IOException {
-    String transcriptXMLUrl = XML_URL_TEMPLATE + videoId;
+    final URL url;
```
Comment (amuamushu) [inline src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java:91]: nit: would it be possible to move the URL building code to its own method?

Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java]: Done, take another look

File: src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java [hunk 2]
```diff
@@ -33,7 +36,10 @@
 
 /** Parses the transcript XML and stores the lines in datastore. */
 public final class TranscriptParser {
-  private static final String XML_URL_TEMPLATE = "http://video.google.com/timedtext?lang=en&v=";
+  private static final String API_URL = "http://video.google.com/timedtext";
```
Comment (billie-cat) [inline src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java:39]: What is this video? Can you add a video?

File: src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java [hunk 3]
```diff
@@ -72,28 +78,36 @@ public static synchronized TranscriptParser getParser() {
...
   }
 
+  private URL getTranscriptUrlForVideo(String videoId) throws IOException {
+    try {
+      URIBuilder urlBuilder = new URIBuilder(API_URL);
+      urlBuilder.addParameter(API_PARAM_LANG, API_LANG_ENGLISH);
+      urlBuilder.addParameter(API_PARAM_VIDEO, videoId);
+      return urlBuilder.build().toURL();
+    } catch (URISyntaxException | MalformedURLException e) {
+      throw new IOException(e.getCause());
```
Comment (billie-cat) [inline src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java:93]: Why are you throwing a different type of exception?

File: src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java [hunk 4]
```diff
@@ -72,28 +78,36 @@ public static synchronized TranscriptParser getParser() {
...
-      Document document = documentBuilder.parse(new URL(transcriptXMLUrl).openStream());
+      Document document = documentBuilder.parse(url.openStream());
       document.getDocumentElement().normalize();
-      return Optional.of(document);
+      return document;
     } catch (ParserConfigurationException | SAXException e) {
-      // TODO: Alert the user.
-      System.out.println("XML parsing error");
-      return Optional.empty();
+      throw new IOException(e.getCause());
```
Comment (billie-cat) [inline src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java:110]: Here too

File: src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java (hunk 2 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java:39]: This isn't a video, this is the `timedtext` API, which we use for our transcript.

File: src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java (hunk 3 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java:93]: https://effectivejava.googleplex.com/third-edition#page=323

File: src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java (hunk 4 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java:110]: https://effectivejava.googleplex.com/third-edition#page=323

File: src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java (hunk 2 above)
Comment (billie-cat) [reply src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java:39]: All your variables are have API_, but it isn't mentioned what the API is maybe add a comment?

Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java:42]: Added the API name to the class javadoc

Comment (billie-cat) [reply src/main/java/com/googleinterns/zoomtube/transcriptParser/TranscriptParser.java:42]: That is good, not everyone reads the class-wish JavaDoc so maybe just add a comment above API_URL?

**Pull request merged**